### ISSUE TITLE
Nested task_spawn with block_on is not working

### ISSUE THREAD

**Issue by manikawnth** [async-rs/async-std#760]
Nested `block_on` works:
```rust
task::block_on(async {
    task::block_on(fib_async(45));
});
```

Nested `task_spawn` with `await` works:
```rust
let s = task::spawn( async {
    task::spawn(fib_async(45)).await;
});
    
task::block_on(s);
```
but nested `task_spawn` with `block_on` **doesn't work**:

```rust
let s = task::spawn( async {
    let s1 = task::spawn(fib_async(45));
    task::block_on(s1);
});
    
task::block_on(s);
```
**Version:**
`async-std = "1.5.0"`

Not sure if I'm doing something obviously wrong!

**Comment by skade** [comment]
Could you please clarify what "doesn't work"? Do you observe a panic or a lock?

**Comment by manikawnth** [comment]
Either it's a lock or an infinite loop. The terminal hangs forever.

This is the full example
```rust
use async_std::task;

fn main() {
    let t = task::spawn(async {        
        println!("parent task spawned");
        let nested_t = task::spawn(async {
            println!("nested task spawned");
        });        
        task::block_on(nested_t); //this doesn't return
        //nested_t.await;   //this yields
    });    
    task::block_on(t);    
}

```

**Comment by dignifiedquire** [comment]
`block_on` inside spawn should really be detected, it doesn't make much sense asfaiu. If you want to block, you can use `.await`

**Comment by Licenser** [comment]
I'm not sure "doesn't make sense" applies here. Yes in the above, simplified, for example it seems nonsensical but in most applications we don't have control over every library that is used or it is possible for some code path to be synchronous for other reasons.

So if a library uses block_on anywhere in their code it becomes unusable in asynchronous code which is a bit of a bummer.

And the same applies to block_on in a block_on too.

**Comment by dignifiedquire** [comment]
Fixed in #799
